test/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java

```diff
@@ -333,15 +340,56 @@ public void testCrNocr() {
 	/* test options in runnable jar files */
 	
 	public void testJarArguments() {
-		String cmdlineArg = "-DcmdlineArg";
-		String testJar = System.getProperty("org.openj9.test.vmArguments.testJar");
-		URL jarUrl = ClassLoader.getSystemClassLoader().getResource("vmargs.jar");
-		assertNotNull("vmargs.jar not found", jarUrl);
-		testJar = jarUrl.getFile();
-		if (isWindows() && testJar.startsWith("/")) {
-			testJar = testJar.substring(1); /* annoying habit of getResource() to put a / at the front of the path */
+		URL jarUrl = ClassLoader.getSystemClassLoader().getResource(VMARGS_JAR);
+		assertNotNull(VMARGS_JAR+" not found", jarUrl);
+		Path jarPath = trimLeadingSlash(jarUrl);		
+		ProcessBuilder pb = makeProcessBuilder(null, new String[] {cmdlineArg, "-jar", jarPath.toString()}, null);
+		Map<String, String> env = pb.environment();
+		String javaToolOptionsArg = "-DjavaToolOptionsArg";
+		env.put(JAVA_TOOL_OPTIONS, javaToolOptionsArg );
+		String ibmJavaOptionsArg = "-DibmJavaOptionsArg";
+		env.put(IBM_JAVA_OPTIONS, ibmJavaOptionsArg );
+		ArrayList<String> actualArguments = runAndGetArgumentList(pb);
+		String fooBar = "-Dfoo=bar";
+		String longProp = "-Da.long.system.property=this is a long system property value to demonstrate long JVM arguments in the manifest file";
+		HashMap<String, Integer> posns = checkArguments(actualArguments, new String[] {javaToolOptionsArg, ibmJavaOptionsArg, cmdlineArg, fooBar,
+		longProp});
+		assertTrue("missing argument: "+fooBar, posns.containsKey(fooBar));
+		assertTrue("missing argument: "+longProp, posns.containsKey(longProp));
+		/* environment variables should come after implicit arguments */
+		int toolsPosn = posns.get(javaToolOptionsArg);
+		int ibmPropPosn = posns.get(ibmJavaOptionsArg);
+		int fooBarPropPosn = posns.get(fooBar);
+		int longPropPosn = posns.get(longProp);
+		Integer cmdlineArgPosn = posns.get(cmdlineArg);
+		assertTrue("Wrong order of properties", ibmPropPosn == toolsPosn + 1);
+		assertTrue("Wrong order of properties", fooBarPropPosn < ibmPropPosn);
+		assertTrue("Wrong order of properties", fooBarPropPosn < cmdlineArgPosn);
+		assertTrue("Wrong order of properties", longPropPosn == fooBarPropPosn + 1);
+	}
+
+	/**
+	 * Test getting VM arguments from a JAR file prefixed by a shell script.
+	 */
+	public void testExecutableJarArguments() {
+		final String HWVMARGS = "hwvmargs.jar";
+		URL jarUrl = ClassLoader.getSystemClassLoader().getResource(VMARGS_JAR);
+		assertNotNull(HWVMARGS+" not found", jarUrl);		
+		Path jarPath = trimLeadingSlash(jarUrl);		
+		
+		try {
+			File outFile = new File(HWVMARGS);
+			outFile.delete();
+			FileOutputStream outStream = new FileOutputStream(outFile);
+			outStream.write(HELLO_WORLD.getBytes());
+			Files.copy(jarPath, outStream);
+			outStream.close();
+		} catch (IOException e) {
```

Comment:
If the exception is thrown, then the test will fail and the stack trace will be available unless we're using a custom test runner of some sort?

This code isn't necessary - let the test throw and the built in mechanisms deal with unexpected exceptions.